client/gutenberg/extensions/presets/jetpack/utils/refresh-registrations.js

```diff
@@ -46,7 +49,18 @@ export default function refreshRegistrations() {
 
 			if ( available && ! registered ) {
 				registerBlockType( blockName, settings );
+				if ( children ) {
+					children.forEach(
+						( { name: childName, settings: childSettings } ) =>
+							void registerBlockType( `jetpack/${ childName }`, childSettings )
+					);
+				}
 			} else if ( ! available && registered ) {
+				if ( children ) {
+					children.forEach(
+						( { name: childName } ) => void unregisterBlockType( `jetpack/${ childName }` )
```

Comment:
We need to be careful about this since. Since if the block is being used somewhere else this could introduce bugs. 
I think we don't have to unregister the inner blocks. They will not show up i the selector if the container block in which the reside is not available.

Reply:
> We need to be careful about this since. Since if the block is being used somewhere else this could introduce bugs.

Oh, like when a block can have multiple parents? Hmm, good point.

> I think we don't have to unregister the inner blocks. They will not show up i the selector if the container block in which the reside is not available.

But that might cause duplicated registration when re-registering a parent block :slightly_frowning_face: Maybe I can add a check for a blocks parent's upon unregistering, to make sure that it has no other potential parents.

Reply:
2563f12